src/InputGroup/InputGroup.js

```diff
@@ -1,272 +1,79 @@
-                            ) : (
-                                <span className={addonClasses}>
-                                    {glyph ? (
-                                        <Icon
-                                            disableStyles={disableStyles}
-                                            glyph={glyph}
-                                            role='presentation' />
-                                    ) : (
-                                        addon
-                                    )}
-                                </span>
-                            )}
-                            <FormInput
-                                {...inputProps}
-                                className={inputClasses}
-                                compact={compact}
-                                disableStyles={disableStyles}
-                                id={inputId}
-                                name={inputName}
-                                onChange={this.handleTextChange}
-                                value={this.state.value} />
-                        </div>
-                    );
-                } else {
-                    return (
-                        <div
-                            {...props}
-                            className={inputGroupClasses}>
-                            <FormInput
-                                {...inputProps}
-                                className={inputClasses}
-                                compact={compact}
-                                disableStyles={disableStyles}
-                                id={inputId}
-                                name={inputName}
-                                onChange={this.handleTextChange}
-                                value={this.state.value} />
-                            {actions ? (
-                                <span className={addonClasses}>
-                                    {children}
-                                </span>
-                            ) : (
-                                <span className={addonClasses}>
-                                    {glyph ? (
-                                        <Icon
-                                            disableStyles={disableStyles}
-                                            glyph={glyph}
-                                            role='presentation' />
-                                    ) : (
-                                        addon
-                                    )}
-                                </span>
-                            )}
-                        </div>
-                    );
-                }
-            }
-        }
+        return (
+            <div
+                {...props}
+                className={inputGroupClasses}
+                compact={compact}>
+                {children}
+            </div>
+        );
     }
 }
 
+InputGroup.Addon = InputGroupAddon;
+InputGroup.Input = FormInput;
```

Comment:
Is `InputGroup.Input` necessary? It looks like the classes get added by checking for FormInput so we don't need to rename it here.

Reply:
Not only is it not necessary, it _shouldn't_ be here.  Keep in mind, the "input" here also needs to support a `FormTextarea`.